# This is the culprit that removes `-fPIC` flag for i686 build.

--- a/hadrian/src/Settings/Packages.hs
+++ b/hadrian/src/Settings/Packages.hs
@@ -479,7 +479,7 @@
 speedHack :: Action Bool
 speedHack = do
     i386   <- anyTargetArch [ArchX86]
-    goodOS <- not <$> anyTargetOs [OSSolaris2]
+    goodOS <- not <$> anyTargetOs [OSSolaris2, OSLinux]
     return $ i386 && goodOS
 
 -- See @rts/ghc.mk@.
